File: docx_parser_converter_ts/src/docx_to_txt/converters/runConverter.ts
import type { Run } from '../../docx_parsers/models/paragraphModels';

/**
 * Converts a run to plain text for TXT output, preserving the logic of the Python implementation.
 */
export class RunConverter {
    /**
     * Convert a run to plain text.
     * @param run The run object.
     * @returns Plain text representation of the run.
     */
    public static convertRun(run: Run): string {
        let runText = '';
        for (const content of run.contents) {
            if (content.type === 'tab') {
                runText += '\t';
            } else if (content.type === 'text') {
                runText += content.text;
            }
        }
        return runText;
    }
} 

File: docx_parser_converter/docx_to_txt/converters/run_converter.py
from docx_parser_converter.docx_parsers.models.paragraph_models import Run, TextContent, TabContent, ImageContent

class RunConverter:
    """
    Class to convert runs to plain text.
    """

    @staticmethod
    def convert_run(run: Run) -> str:
        """
        Convert a run to plain text.

        Args:
            run (Run): The run object.

        Returns:
            str: Plain text representation of the run.

        Example:
            .. code-block:: python

                run = Run(
                    contents=[
                        RunContent(run=TextContent(text="Hello")),
                        RunContent(run=TabContent(type="tab")),
                        RunContent(run=TextContent(text="world"))
                    ],
                    properties=RunStyleProperties()
                )
                run_text = RunConverter.convert_run(run)
                print(run_text)  # Output: "Hello\tworld"
        """
        run_text = ""
        for content in run.contents:
            if isinstance(content.run, TabContent):
                run_text += "\t"
            elif isinstance(content.run, TextContent):
                run_text += content.run.text
            elif isinstance(content.run, ImageContent):
                run_text += RunConverter.convert_image(content.run)
        return run_text

    @staticmethod
    def convert_image(image: ImageContent) -> str:
        """
        Converts an ImageContent to a plain text placeholder.

        Args:
            image (ImageContent): The image content to convert.

        Returns:
            str: A plain text placeholder for the image.

        Example:
            The output might look like:

            .. code-block:: text

                [Image: Picture 1]
                [Image]
        """
        if image.title:
            return f"[Image: {image.title}]"
        elif image.alt_text:
            return f"[Image: {image.alt_text}]"
        else:
            return "[Image]"
